src/PHPStan/PregReplaceCallbackClosureTypeExtension.php

```diff
@@ -61,6 +61,18 @@ public function getTypeFromStaticMethodCall(MethodReflection $methodReflection,
             $matchesType = new ConstantArrayType(
                 $matchesType->getKeyTypes(),
                 array_map(static function (Type $valueType): Type {
+                    if (count($valueType->getConstantArrays()) === 1) {
```

Comment:
when `PREG_OFFSET_CAPTURE` the `$valueType` is again a constant-array-type. 

`TypeCombinator::removeNull` does not deep iterate into arrays

Reply:
I used the same logic already arround, but I guess we will get problems in the future with it when patterns are used which return more then 1 constant array (maybe one of the tagged union cases)

Reply:
we could use https://github.com/phpstan/phpstan-src/blob/1.12.x/src/Type/TypeTraverser.php instead